Login with contract

stevenvegt opened the pull request:
Explain how Nuts is going to authenticate users. It contains a list of requirements, some examples and a basic flow.

Question to the reviewers: What is missing, do you agree with the current setup?

stevenvegt committed:
Add initial version of login with consent page

stevenvegt committed:
Added IRMA login diagram

woutslakhorst commented on pages/login-contract.rst:
```diff
@@ -0,0 +1,79 @@
+
+So, we have several parties:
+* A user, probably a care professional.
+* A trusted registry which administres traits about the user such as an identity or medical qualification
```
typo

woutslakhorst replied:
Not rendered as bullets since ReST requires indentation

woutslakhorst commented on pages/login-contract.rst:
```diff
@@ -0,0 +1,79 @@
+* The users software own software system
+* Another system containing data the user wants to access.
+
+The consent should:
```
must

woutslakhorst commented on pages/login-contract.rst:
```diff
@@ -0,0 +1,79 @@
+
+You can try this simple example over here: https://regex101.com/r/sHfsKq/1
+
+Matching groups
```
The why per group is also important to mention: some parts are needed for security, others for matching consent records and the rest for limiting scope.

woutslakhorst commented on pages/login-contract.rst:
```diff
@@ -0,0 +1,79 @@
+-----
+We can encode our *scope* field based on the FHIR consent scope set: https://www.hl7.org/fhir/valueset-consent-scope.html
+
+IRMA Contracts
```
Maybe already mention that any identity scheme capable of signing with attributes will be ok?

stevenvegt committed:
Improves the login contract docs

Fixed feedback
Improved the contracts
Added extra example
Emphasize that IRMA is not the only possible solution

stevenvegt committed:
Split lines on the end of a sentence.
